packages/terra-avatar/src/Avatar.jsx

```diff
@@ -43,56 +44,89 @@ const defaultProps = {
   variant: AvatarVariants.USER,
 };
 
-const Avatar = ({
-  alt,
-  ariaLabel,
-  image,
-  initials,
-  variant,
-  ...customProps
-  }) => {
-  // Checks to run when not in production
-  if (process.env.NODE_ENV !== 'production') {
-    if (image && initials) {
-      // eslint-disable-next-line
-      console.warn('Only one of the props: [image, initials] should be supplied.');
-    }
+class Avatar extends React.Component {
+  static generateImagePlaceholder(variant) {
+    const avatarIconClassNames = cx([
+      'avatar-icon',
+      variant,
+    ]);
+
+    return <span className={avatarIconClassNames} />;
   }
 
-  const attributes = Object.assign({}, customProps);
+  static generateImage(image, variant, alt) {
+    const icon = Avatar.generateImagePlaceholder(variant);
 
-  const avatarClassNames = cx([
-    'avatar',
-    attributes.className,
-  ]);
+    return <TerraImage className={cx('avatar-image')} src={image} placeholder={icon} alt={alt} />;
+  }
 
-  const avatarIconClassNames = cx([
-    'avatar-icon',
-    variant,
-  ]);
+  constructor(props) {
+    super(props);
 
-  const avatarTextClassNames = cx([
-    { 'avatar-text-small': initials && initials.length === 3 },
-    { 'avatar-text-large': initials && initials.length === 2 },
-  ]);
+    // If image has been provided we need to generate the image to display and store it in the state
+    if (props.image) {
+      const { alt, image, variant } = props;
 
-  const icon = <span className={avatarIconClassNames} />;
+      const imageComponent = Avatar.generateImage(image, variant, alt);
+      this.state = { imageComponent };
+    }
 
-  let avatarContent;
-  if (image) {
-    avatarContent = <TerraImage className={cx('avatar-image')} src={image} placeholder={icon} alt={alt} />;
-  } else if (initials && (initials.length === 2 || initials.length === 3)) {
-    avatarContent = <span className={avatarTextClassNames}>{initials.toUpperCase()}</span>;
-  } else {
-    avatarContent = icon;
+    // Checks to run when not in production
+    if (process.env.NODE_ENV !== 'production') {
+      if (props.image && props.initials) {
+        // eslint-disable-next-line
+        console.warn('Only one of the props: [image, initials] should be supplied.');
+      }
+    }
   }
 
-  return (
-    <div {...attributes} aria-label={ariaLabel} className={avatarClassNames}>
-      {avatarContent}
-    </div>
-  );
-};
+  componentWillReceiveProps(newProps) {
+    // If we have an image to display (they take precedence) and one of its attributes have changed
+    if (newProps.image && (newProps.image !== this.props.image ||
+      newProps.alt !== this.props.alt || newProps.variant !== this.props.variant)) {
+      const { alt, image, variant } = newProps;
```

Comment:
Nit: for reusability, would it be better this into a static function
```javascript
//not sure on the function name
static <function> (props) {
  const { alt, image, variant } = props;
  const imageComponent = Avatar.generateImage(image, variant, alt);
  this.setState({ imageComponent });
}
//call it in both places
Avatar.function(newprops);
```
This way we won't repeat the logic twice.

Reply:
Calling `this.setState({ imageComponent });` from the constructor is a runtime error and reassigning state after construction is bad practice, so the most a static function for this could do is extract the individual variables from props. If you still think I should go ahead and make a static method for it, I'll do so.

Reply:
ahhhh missed that, good catch 👍